### PR TITLE
Adds Necromatic Sect

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request

Adds a Necromantic Sect to the Chaplain's sect choices
You can offer up non-sentient living creatures to gain favor (200 + their current health, so the healthier the better)
Or for a more passive way, offering up organs for 10 favor each.
You can then spend that favor on either summoning a high functioning zombie/skeleton from the observer pool,
or spend it on resurrecting someone (which is much more expensive)

Raise Undead - Costs 1500 favor - Takes 90 seconds
Raise Dead - Costs 2500 favor - Takes 120 seconds
Living Sacrifice - Gives 200+health favor for non-sentient creatures - Takes 60 seconds
Organs - Gives 10 favor per organ

You cant resurrect someone if they arent in the observer pool, or in their body.

## Why It's Good For The Game

Yet another way to keep players in-round, be it as a brand new high functioning, non infective, zombie
or as a full resurrection.

As far as I can tell, without any admin responses to prayers, the Chaplain has almost nothing going on during most rounds.
Occasinally they get funerals, or get to fight a cult or revenant, and theres the flame and tech sects too, but this brings a bit more to the table, and hopefully crew will come by the chapel more often to interact with the chaplain, especially if medbay gets nuked.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

Screenshot_9

Screenshot_10

Screenshot_1

Screenshot_2

Screenshot_3

Screenshot_4

Screenshot_6

Screenshot_8

Screenshot_1

Screenshot_2

Screenshot_3

Screenshot_4

Screenshot_2

</details>

## Changelog
:cl:
add: The Chaplain now has access to a Necromancy Sect!
add: Living Sacrifice rite for the Necromancy Sect, Sacrifice Non-Sentient living creatures for favor
add: Raise Dead rite for the Necromancy Sect, Resurrect anyone with the power of god - 2500 Favor cost
add: Raise Undead rite for the Necromancy Sect, Raise a high functioning zombie/skeleton from the observer pool - 1500 Favor cost
add: Necromancy Sect uses the green altar
add: Summoned Undead are sub-chaplains, and have their own clothing and bible!
/:cl:

### CODE REVIEW

**Comment by Miliviu** [comment]
I think I would raise the favor gained from sacrificing sentient creatures and lower the favor gained from sacrificing non-sentient creatures. Idk, what do you think? :)

**Comment by Fronsis** [comment]
Maybe the Chapel now should include an empty iv and empty bloody bag? I just hope this doesn't end up making Chaplains stealing the blood freezer lol

What's the aproximate timer for a Chaplain to get those 2500 points to revive someone?

Good idea though!

**Comment by ro5490** [comment]
> Maybe the Chapel now should include an empty iv and empty bloody bag? I just hope this doesn't end up making Chaplains stealing the blood freezer lol
> 


Blood bags get deleted on offering, so the IV wouldnt help really, you'd have to order blood crates from cargo to get a decent amount

> What's the aproximate timer for a Chaplain to get those 2500 points to revive someone?
> 
Depends, if they rushed medbay bloodbags roundstart they'd be able to get roughly 2200 favor
But I doubt medbay would let the chaplain into surgery to get the blood bags in the first place, but lets say they did
Roundstart 2200~ takes about what, 2 minutes to run and grab the bags and come back, then you've still got to find a sacrifice, lets say a full health monkey, so thats another 300 favor.
Say that takes roughly 3 minutes to get and buckle, as monkeys can be tricky sometimes and its got to be alive to count.
so, 5 minutes to get the favor, and another 2 to Raise Dead, Roughly 7 minutes?
but blood bags get hard deleted on offering, and I highly doubt medbay would be okay with losing so many blood-bags as they can cost quite a bit from cargo.

If they go via the sacrifice route, using non-sentient things like monkeys and pets those take roughly 1 minute per sacrifice, pets only give between 200 and 300 favor, except for frank who'd give 2700 due to his health pool. and monkeys or similar would give roughly 300 favor
So lets go with the regular of 300 per simplemob, it'd take rougly 8 sacrifices, so 8 minutes. but you'd also have to go about and find those simplemobs, or a monkey box and the chaplain doesnt really have much access around the station to do so.

so lets say it takes the chap like 15 minutes to find enough simplemobs, then another 8 to sacrifice them all
thats about 23 minutes, then another 2 to cast the Raise Dead rite
So roughly 25 minutes, for ONE resurrection

OR
3 full health sentient sacrifices, so roughly 3000 favor
say that takes about 10-15 mintes to get willing people, (or unwilling if you're an antag)
plus another minite per sac, so lets say 18 minutes.
then 2 minutes for the res, so roughly 15-20 minutes for a single resurrection
And dont forget it petty much shards someone if they get sacrificed, so all their gear, items, everything. gone.


if you're going for Zombies however, that'd cut the times down by about 0.4, so 25 minutes becomes 15, and 15 min becomes 6
roughly
because they cost 1000 less to summon.


Tl;Dr-
Roughly 25 mintes for One Resurrection
or 15 if using sentient mobs
15 for Zombies
or 6 if using sentient mobs

**Comment by Miliviu** [comment]
Yeah, I think it's balanced, I am concerned that the Chaplains are going to Tide to get things to offer. But idk, it sounds fun if they don't!

**Comment by ro5490** [comment]
Currently in the process of adding clothing to the summoned zombies, possibly making them sub-chaplains?
And adding some "magical" effects to the summoning ritual

**Comment by EvilDragonfiend** [comment]
It would be fine as long as we can get other empty blood bags from somewhere, like printing them from a techfab with a cost of plastic.

**Comment by ro5490** [comment]
> It would be fine as long as we can get other empty blood bags from somewhere, like printing them from a techfab with a cost of plastic.

The whole point is that they're hard to come by, otherwise people could just print a bunch and fill them up. Easy large amounts of favor for full resurrections.
You can order them from cargo for cash though

**Comment by Sapfearon** [comment]
tech cult currently completly broken by the way, so chaplain only has fire cult as right now. So new cult is welcomed, for sure.
however, do we really need zombies? why not spooky skeletons instead?

**Comment by ro5490** [comment]
> tech cult currently completly broken by the way, so chaplain only has fire cult as right now. So new cult is welcomed, for sure. however, do we really need zombies? why not spooky skeletons instead?

Tempted to make it randomly pick either zombie or skeleton tbh, but either way the zombies are non-infectious

Whats broken with tech-cult? I'll see if I can fix it

**Review by PowerfulBacon — COMMENTED** [review]
I dislike that it can delete sentient mobs entirely and uses blood bags which may or may not result in stealing from places that need them

**Comment by ro5490** [comment]
> I dislike that it can delete sentient mobs entirely and uses blood bags which may or may not result in stealing from places that need them

I'll work on this tomorrow, and possibly change blood bags for hearts or organs in general and lower the amount of favour gained for those items.
I'll also look into maybe just calling gib()  instead of deleting them, or something similar that means they can't be revived

**Comment by ro5490** [comment]
Removed the ability to sacrifice sentient beings for reasons

added the possibility to summon a skeleton instead of a zombie
summoned undead now are named "Holy Undead (number)"
Changed blood bags for organs, and lowered the favor gained

**Comment by ro5490** [comment]
Wiki edit is saved on notepad and ready to go

**Comment by ro5490** [comment]
Currently putting this on hold until the decision on 
https://github.com/BeeStation/BeeStation-Hornet/pull/7194
Is concluded.
As I might need to re-do this a bit differently due to the changes

File: code/modules/religion/rites.dm [hunk 1]
```diff
@@ -218,3 +216,168 @@
...
+///the target
+	var/mob/living/carbon/human/raise_target
+/datum/religion_rites/raise_dead/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
+	if(!LAZYLEN(movable_reltool.buckled_mobs))
```
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	if(!length(movable_reltool.buckled_mobs))
```
Don't use `LAZYLEN` for non lazy lists.

File: code/modules/religion/rites.dm [hunk 2]
```diff
@@ -218,3 +216,168 @@
...
+//the living creature chosen for the sacrifice of the rite
+	var/mob/living/chosen_sacrifice
+/datum/religion_rites/living_sacrifice/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
+	if(!LAZYLEN(movable_reltool.buckled_mobs))
```
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	if(!length(movable_reltool.buckled_mobs))
```

File: code/modules/religion/rites.dm [hunk 3]
```diff
@@ -218,3 +216,168 @@
...
+	"... come forth from the beyond ...",
+	"... fresh life awaits you ...",
+	"... return to us ...",
+	"... by the power granted by the gods ...",
+	"... you shall rise again ...")
+	invoke_msg = "Welcome back to the mortal plain."
+	favor_cost = 2500
+
+///the target
+	var/mob/living/carbon/human/raise_target
```
Comment (ivanmixo) [inline code/modules/religion/rites.dm:319]:
```suggestion
	var/mob/living/carbon/human/raise_target

```

Spacing

File: code/modules/religion/rites.dm [hunk 4]
```diff
@@ -218,3 +216,168 @@
...
+		if(raise_target.stat != DEAD)
+			to_chat(user, "<span class='warning'>You can only resurrect dead bodies, this one is still alive!</span>")
+			raise_target = null
+			return FALSE
+		if(!raise_target.mind)
+			to_chat(user, "<span class='warning'>This creature has no connected soul...")
+			raise_target = null
+			return FALSE
+		else
+			raise_target.notify_ghost_cloning("Your soul is being summoned back to your body by mystical power!", source = src)
```
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	for(var/mob/living/carbon/r_target in movable_reltool.buckled_mobs)
		if(!iscarbon(r_target))
			to_chat(user, "<span class='warning'>Only carbon lifeforms can be properly resurrected!</span>")
			return FALSE
		if(r_target.stat != DEAD)
			to_chat(user, "<span class='warning'>You can only resurrect dead bodies, this one is still alive!</span>")
			return FALSE
		if(!r_target.mind)
			to_chat(user, "<span class='warning'>This creature has no connected soul...")
			return FALSE
		raise_target = r_target
		raise_target.notify_ghost_cloning("Your soul is being summoned back to your body by mystical power!", source = src)
```

If this doesn't compile/work for whatever reason yell at me.

File: code/modules/religion/rites.dm [hunk 5]
```diff
@@ -218,3 +216,168 @@
...
+
+//the living creature chosen for the sacrifice of the rite
+	var/mob/living/chosen_sacrifice
+/datum/religion_rites/living_sacrifice/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
```
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	if(!religious_tool || !ismovable(religious_tool))
		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
		return FALSE
	var/atom/movable/movable_reltool = religious_tool
```

File: code/modules/religion/religion_sects.dm [hunk 6]
```diff
@@ -216,3 +216,28 @@
...
+	rites_list = list(/datum/religion_rites/raise_dead, /datum/religion_rites/living_sacrifice, /datum/religion_rites/raise_undead)
+	altar_icon_state = "convertaltar-green"
+
+//Necro bibles don't heal or do anything special apart from the standard holy water blessings
+/datum/religion_sect/necro_sect/sect_bless(mob/living/blessed, mob/living/user)
+	return TRUE
+
+/datum/religion_sect/necro_sect/on_sacrifice(obj/item/N, mob/living/L) //and this
+	var/obj/item/organ/organ = N
+	if(!istype(organ)) //how...
```
Comment (ivanmixo) [inline code/modules/religion/religion_sects.dm]: Very easily, it can take any item. Just because you typecasted it to be an organ doesn't mean it actually is an organ. ;3

File: code/modules/religion/rites.dm [hunk 7]
```diff
@@ -218,3 +216,168 @@
...
+
+///the target
+	var/mob/living/carbon/human/raise_target
+/datum/religion_rites/raise_dead/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
```
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	if(!religious_tool || !ismovable(religious_tool))
		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
		return FALSE
	var/atom/movable/movable_reltool = religious_tool
```

File: code/modules/religion/religion_sects.dm (hunk 6 above)
Comment (ro5490) [reply code/modules/religion/religion_sects.dm]: Rip, I missed that, thanks

**Comment by ro5490** [comment]
Updating this to the TGUI version.
This PR now requires https://github.com/BeeStation/BeeStation-Hornet/pull/7194

**Comment by ro5490** [comment]
or should do, at least

**Comment by ro5490** [comment]
If not, I'll just re-PR these when that gets merged, got backup copies saved

**Comment by ro5490** [comment]
> > > tech cult currently completly broken by the way, so chaplain only has fire cult as right now. So new cult is welcomed, for sure. however, do we really need zombies? why not spooky skeletons instead?
> > 
> > 
> > Tempted to make it randomly pick either zombie or skeleton tbh, but either way the zombies are non-infectious
> > Whats broken with tech-cult? I'll see if I can fix it
> 
> tech cult currently gives you synthetic race...with human kapu limbs. Which means you can't repair them, you can't heal them, you can't even stop bleeding. The only workaround is after race change get full augments but that's kinda ruins whole point. Oh and best and pretty much only way of earning favors is using....botany potato cells. Which kinda ruins whole "techno" part of cult. And if you want to use bluespace cells - you need a insane amount of them just for 1! conversion.


Fixed this with https://github.com/BeeStation/BeeStation-Hornet/pull/7197




> About Necro sect - don't make favor earning too slow. If you spent half of round for just 1 gimmick race change for single person it's honestly not worth it. Remember that you can find spooky skele in maints and mass clone it while doing brain transplants. Fire cult currently allows to earn 1200 favors per body, which is...2 or 3 items fireproofed? Pretty fast if you have good burn chamber/hand of god + cubes on demand. And even than, you barely get 5 people fireproofed in 2 hours long shift despite doing PR non-stop about it. So faster you can show people some results, better. Naturally, reviving should be overpriced so at best you can do it 1-2 times per shift.
> 
> My biggest problems are timers on every sacrifice/ritual. Holy hell, 120 seconds? 60 seconds already long enough i think.

True, I'll probably adjust this after a testmerge and get general feedback on it. I might buff the amount of favor gained by organ sacrifices, or if Bacon approves, I'll bring back the massive favor gain from sacrificing sentient creatures. If not, then I'll reduce the sacrifice time for sure

**Comment by ro5490** [comment]
This should be good to go!

**Review by PowerfulBacon — APPROVED** [review]
